### PR TITLE
adding additional 'make static' command 

### PR DESCRIPTION
**Pull request by endophage**
Just creates static builds of server and signer for reliable use with alpine container for official images.

I'm by no means good at Makefiles so let me know if there's a better way to do this. 

Signed-off-by: David Lawrence <david.lawrence@docker.com> (github: endophage)

### CODE REVIEW

**Comment by cyli** [comment]
Non-blocking - you can't seem to compile the binary statically in OS X - I'm wondering if it'd be worth it to wrap those in a
```
ifeq ($(UNAME_S),Darwin)
        ...
endif
```

Alternately we can just document in a comment that they don't work in OS X.  :)

LGTM otherwise!  The server compiles just fine.  Signer not so much because of CGO.

**Comment by endophage** [comment]
@cyli because the SoftHSM stuff has been taken out, the signer shouldn't have any CGo dependencies. Will need to look at that.

**Comment by riyazdf** [comment]
I second @cyli's comment about notary-signer on OS X (just tried it myself).  Otherwise LGTM!

**Comment by endophage** [comment]
Good to know. I'm considering this blocked until we work out why signer still has CGo dependencies.

**Comment by cyli** [comment]
I'm wondering if the +pkcs11 build directive has something to do with it (i'll check when I get home)

**Comment by cyli** [comment]
Nope.  :(   that wasn't it, sorry.  Server seems to be failing for me now too?

**Comment by cyli** [comment]
Looks like maybe you just can't pass the `-static` flag in OS X?  (https://stackoverflow.com/questions/3801011/ld-library-not-found-for-lcrt0-o-on-osx-10-6-with-gcc-clang-static-flag)

I'm getting that error above:

```
+ /Users/cyli/Go/src/github.com/docker/notary/bin/static/notary-server
# github.com/docker/notary/cmd/notary-server
/usr/local/Cellar/go/1.5.2/libexec/pkg/tool/darwin_amd64/link: running clang failed: exit status 1
ld: library not found for -lcrt0.o
clang: error: linker command failed with exit code 1 (use -v to see invocation)

godep: go exit status 2
make: *** [/Users/cyli/Go/src/github.com/docker/notary/bin/static/notary-server] Error 1  
```

**Comment by endophage** [comment]
@cyli well that's a bit rubbish :-(

I'll wrap it with a conditional as you suggested. Technically we only need to statically build for linux as this is just for building official binaries to be run in linux containers.

**Comment by endophage** [comment]
@cyli can you check that it now outputs the notifications on mac?

**Comment by riyazdf** [comment]
Just tried on my mac, I didn't get the warnings:

```
~ notary $ make static
pkcs11 import was not found anywhere without a build tag, yay

+ /Users/c1mqm40sg944/go/src/github.com/docker/notary/bin/static/notary-server
# github.com/docker/notary/cmd/notary-server
/usr/local/go/pkg/tool/darwin_amd64/link: running clang failed: exit status 1
ld: library not found for -lcrt0.o
clang: error: linker command failed with exit code 1 (use -v to see invocation)

godep: go exit status 2
make: *** [/Users/c1mqm40sg944/go/src/github.com/docker/notary/bin/static/notary-server] Error 1
```

File: Makefile [hunk 1]
```diff
@@ -71,6 +71,22 @@ ${PREFIX}/bin/notary-signer: NOTARY_VERSION $(shell find . -type f -name '*.go')
 	@echo "+ $@"
 	@godep go build -tags ${NOTARY_BUILDTAGS} -o $@ ${GO_LDFLAGS} ./cmd/notary-signer
 
+ifeq ($(UNAME_S),Darwin)
```
Comment (cyli) [inline Makefile]: changing this to `shell uname -s` worked for me

Comment (endophage) [inline Makefile]: Updated :-)

**Comment by cyli** [comment]
LGTM!

**Pull request merged**